src/averageAges.js

```diff
@@ -14,12 +14,22 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const selection = people.filter(
+    man => {
+      const check = man.sex === 'm' && century
+        ? man.sex === 'm' && Math.ceil(man.died / 100) === century
+        : man.sex === 'm';
+
+      return check;
+    });
+
+  const avg = selection.reduce((sumAvg, man) => (
+    man.died - man.born) / selection.length + sumAvg, 0
+  );
```

Comment:
you use this logic a couple of times - let's probably move its logic to a helper function?

Reply:
👌fixed
I've made a separate function `avaregeAge`